### PR TITLE
Graph algo

### PR DESCRIPTION
**Pull request by Suxiaocai**
1. Fixed PageRank template
2. Added shortest path algos
3. Added similarity algos
4. Added install free queries

### CODE REVIEW

**Comment by xchang2014** [comment]
1. Please add Jaccard to the install.sh list. Only asking for graph name is fine.

**Comment by Suxiaocai** [comment]
1. Added Jaccard as gtmp, using same format as cosine similarity
2. Added reverse edge choice for similarity algos 
3. Added uniform weight for cosine similarity
4. Added loop to let user try again when they type wrong edge attribute name

File: graph_algorithms/algorithms/install.sh [hunk 1]
```diff
@@ -153,130 +184,211 @@ while [ !$finished ]; do
 		egs=${egs/,/|}
 		egs="(${egs})"
 	fi
-	sed -i "s/\*edge-types\*/$egs/g" ${templPath}/${algoName}.gsql
+	sed -i "s/\*edge-types\*/$egs/g" ${genPath}/${algoName}.gsql
+
+	# 4.2 Ask for reverse edge type for similarity algos. 
+        if [[ ${algoName} == similarity* ]]; then
+		read -p 'Reverse Edge type: ' reveg
```
Comment (xchang2014) [inline graph_algorithms/algorithms/install.sh]: Should be second hop edge type. not necessarily a reverse edge， right?

File: graph_algorithms/algorithms/generated/shortest_ss_pos_wt_file.gsql [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# Algorithm for Single-Source Shortest Path with positive weights on edges 
+# The path parameter let user choose whether calculate path or not
+
+CREATE QUERY shortest_ss_pos_wt_file (VERTEX v, BOOL display, BOOL path, STRING filepath) FOR GRAPH social {
```
Comment (xchang2014) [inline graph_algorithms/algorithms/generated/shortest_ss_pos_wt_file.gsql]: remove display, path. Print the whole path by default.

File: graph_algorithms/install_free_algos/example/similarity_jaccard_single_attr.gsql [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+This is an example to show how to insert edges between "Person" and store the similarity scores. 
+*/
+
+  SumAccum<int> @intersection_size, @@set_size_A, @set_size_B;
+ 
+    #= input;
+  Start (ANY) = {input};
+    Start = select s
+            from Start:s
+            accum @@set_size_A += s.outdegree(edge_type);
```
[the remark this answers is not in the document]
Comment (Suxiaocai) [reply graph_algorithms/install_free_algos/example/similarity_jaccard_single_attr.gsql]: yes, it works

File: graph_algorithms/algorithms/generated/shortest_ss_pos_wt_file.gsql (hunk 2 above)
Comment (Suxiaocai) [reply graph_algorithms/algorithms/generated/shortest_ss_pos_wt_file.gsql]: Shall l totally remove display for every query? There is also display for queries like Closeness Centrality. Or just keep the display choice for ACCUM version?

**Pull request merged**